VideoLocker/src/main/java/org/edx/mobile/view/MyCourseListTabFragment.java

```diff
@@ -32,13 +35,19 @@
     private boolean refreshOnResume;
 
     @Inject
-    LoginPrefs loginPrefs;
+    private LoginPrefs loginPrefs;
 
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         environment.getSegment().trackScreenView(ISegment.Screens.MY_COURSES);
         EventBus.getDefault().register(this);
+
+        // Restore cache of the courses for which the user has downloaded any videos
+        PrefManager.UserPrefManager prefs = new PrefManager.UserPrefManager(MainApplication.application);
+        if (!prefs.isVideosCacheRestored()) {
+            new RestoreVideosCacheDataTask(MainApplication.application).execute();
+        }
```

Comment:
This logic could be moved to a static method in `RestoreVideosCacheDataTask`, e.g. `RestoreVideosCacheDataTask.executeInstanceIfNeeded()` for better symmetry (i.e. it sets the shared preference value so might as well have it check the value as well). But it's up to you; this is fine as well.